### PR TITLE
[JUJU-1830] [CONTRIBUTING.md] Add CLA section + other small fixes

### PR DESCRIPTION
**Pull request by barrettj12**
Added a section about the CLA to the contribution guide. This should definitively answer all CLA questions once and for all. It's linkable too, so we can just send the link to new contributors.

A few other minor changes/fixes too.

### CODE REVIEW

**Comment by barrettj12** [comment]
Once this merges, we should get jujubot to link to the CLA guide when it does this:
Screenshot from 2022-09-21 14-38-46

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,7 +52,7 @@ Go can be [installed](https://golang.org/doc/install#install) from the official
 
 [Snap](https://snapcraft.io/go) may also be used to install Go on Linux.
 
-    snap install go --channel=1.18/stable --classic
+    snap install go --channel=latest/stable --classic
```
Comment (wallyworld) [inline CONTRIBUTING.md:55]: not latest, best to stick to a version, eg 1.19

**Review by wallyworld — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (barrettj12) [reply CONTRIBUTING.md:55]: I have latest installed lmao

**Pull request merged**